### PR TITLE
JOSS-Review add a discussion about algorithms.

### PR DESCRIPTION
**Pull request by drvinceknight**
Add some timings for an 11 by 11 game but also describe the fact that
the algorithms themselves are the limiting factor due to the complexity.

Addresses #45

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -96,6 +96,31 @@ reference material on the algorithms:
 software is automatically tested using a combination of doc (this paper is also
 tested), unit, integration and property based tests with 100% coverage.
 
+There are potential limitiations associated with Nashpy, these are due to the
```
Comment (Fil) [inline paper.md]: limitations

File: paper.md [hunk 2]
```diff
@@ -96,6 +96,31 @@ reference material on the algorithms:
...
+complexity of the algorithms themselves. For example, support enumeration
+enumerates all potential pairs of strategy. For $n\times n$ sized square
+matrices it has complexity of the order of $\mathcal{O}\left({2^n}^2\right)$.
+All implementations provided in Nashpy ensure these affects are reduced: numpy
```
Comment (Fil) [inline paper.md]: effects

File: paper.md [hunk 3]
```diff
@@ -96,6 +96,31 @@ reference material on the algorithms:
...
+[@scipy] provides C based implementations for vectorized performance.
+Furthermore, all algorithms are generators which ensures that not all equilibria
+must be found before one is returned. For example, below an 11 by 11 game is
+considered and timings computed are shown for relative comparison.  Using the
+more efficient Lemke-Howson algorithm [@lemke1964equilibrium] an equilibria is
```
Comment (Fil) [inline paper.md]: equilibrium

File: paper.md [hunk 4]
```diff
@@ -96,6 +96,31 @@ reference material on the algorithms:
...
+founds approximately 3000 times faster.
```
Comment (Fil) [inline paper.md]: found

File: paper.md [hunk 5]
```diff
@@ -96,6 +96,31 @@ reference material on the algorithms:
...
+
+```
+>>> from pprint import pprint
+>>> A = np.eye(11)
```
Comment (Fil) [inline paper.md]: this is a trivial game :)

Comment (drvinceknight) [reply paper.md]: Thanks for the typos: have addressed in #49 (apologies).

>>> this is a trivial game :)

Yeah! It is in it's construction but in terms of the equilibria is isn't really, for example for `A=np.eye(5)` we still get quite a large range of different equilibria:

```python
>>> A = np.eye(N)
>>> B = A[::-1]
>>> game = nash.Game(A, B)
>>> list(game.support_enumeration())
[(array([ 0.,  0.,  1.,  0.,  0.]), array([ 0.,  0.,  1.,  0.,  0.])),
 (array([ 0.5,  0. ,  0. ,  0. ,  0.5]),
  array([ 0.5,  0. ,  0. ,  0. ,  0.5])),
 (array([ 0. ,  0.5,  0. ,  0.5,  0. ]),
  array([ 0. ,  0.5,  0. ,  0.5,  0. ])),
 (array([ 0.33333333,  0.        ,  0.33333333,  0.        ,  0.33333333]),
  array([ 0.33333333,  0.        ,  0.33333333,  0.        ,  0.33333333])),
 (array([ 0.        ,  0.33333333,  0.33333333,  0.33333333,  0.        ]),
  array([ 0.        ,  0.33333333,  0.33333333,  0.33333333,  0.        ])),
 (array([ 0.25,  0.25,  0.  ,  0.25,  0.25]),
  array([ 0.25,  0.25,  0.  ,  0.25,  0.25])),
 (array([ 0.2,  0.2,  0.2,  0.2,  0.2]),
  array([ 0.2,  0.2,  0.2,  0.2,  0.2]))]
```

An alternative would be to use a random game however that gets interesting as degenerate games could be used in which case the algorithms can become "nonsensical".

:)

Comment (Fil) [reply paper.md]: cool!

**Pull request merged**